### DISCUSSION TITLE
Comparing commits shows duplicates

### DISCUSSION THREAD

**Discussion by steka** [gitextensions/gitextensions#11200] [Q&A] [answered]
Is there a special reason that when comparing commits it show **two** diff lists with the same commit hash.

![image](https://github.com/gitextensions/gitextensions/assets/2719766/da3c3ea0-595d-4e03-8f13-f4147f825bdb)

It is just confusing!

I kind of see it as a bug that should be fixed.

**Comment by mstv** [comment]
> "when comparing commits"

Please be more specific: What is selected in the Browse window? How do you open this dialog? Which version do you use (copy button in About dialog)?

**Comment by steka** [comment]
For example, by right clicking on a commit and selecting "Compare with current branch"

This is the About information:

- Git Extensions 4.1.0.16698
- Build 1fe52a137acef1fbd2b653dea263b44ad5844177
- Git 2.42.0.windows.2
- Microsoft Windows NT 10.0.19045.0
- .NET 6.0.16
- DPI 96dpi (no scaling)
- Portable: False
- Microsoft.WindowsDesktop.App Versions

```
    Microsoft.WindowsDesktop.App 3.1.12 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 5.0.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

**Reply by mstv** [reply]
The dialog is primarily intended for the comparison of _different_ branches, also with their common base commit (merge base).
If an older commit of the _same_ branch is compared with, it is identical to the base commit.
I agree this edge case is not handled well here.
I would rather expect the second diff group to show the diff of the selected commit with the base commit, i.e. "(0) Diff BASE with A f1ac65c0", "no changes".

(I wonder why there is a difference between the Diff window and the Browse window with a multi-selection of the old and the current commit.)

**Comment by steka** [comment]
I do hope that this issue can be "clean up", so it becomes less confusing.

**Reply by mstv** [reply, marked as answer]
https://ci.appveyor.com/project/gitextensions/gitextensions/builds/48008677/artifacts
